api/dao/user_dao.js

```diff
@@ -3,129 +3,135 @@ const Student = require('../../models/Student');
     // Look for tutors with the same email
-    Tutor.find({email: email}, function (err, docs) {
-      if (err) {
-        console.error('Error checking if email is taken:', err);
-        callback(err);
-
-      } else if (docs.length > 0) {
-        // Found a tutor with the same email
-        callback(null, true);
-
-      } else {
-
-        // Look for students with the same email
-        Student.find({email: email}, function (err, docs) {
-          if (err) {
-            console.error('Error checking if email is taken:', err);
-            callback(err);
-
-          } else if (docs.length > 0) {
-            // Found a student with the same email
-            callback(null, true);
-
-          } else {
-            // No student or tutor found with that email!
-            callback(null, false);
-          }
+        Tutor.find({email: email}, function (err, docs) {
+            if (err) {
+                console.error('Error checking if email is taken:', err);
+                callback(err);
+
+            } else if (docs.length > 0) {
+                // Found a tutor with the same email
+                callback(null, true);
+
+            } else {
+
+                // Look for students with the same email
+                Student.find({email: email}, function (err, docs) {
+                    if (err) {
+                        console.error('Error checking if email is taken:', err);
+                        callback(err);
+
+                    } else if (docs.length > 0) {
+                        // Found a student with the same email
+                        callback(null, true);
+
+                    } else {
+                        // No student or tutor found with that email!
+                        callback(null, false);
+                    }
+                });
+            }
         });
-      }
-    });
-  },
-
-  createStudent: function(student, callback) {
-    Student.create(student, function (err, student_instance) {
-      if (err) {
-        console.error('Error creating a new student:', err);
-        callback(err);
-      } else {
-        callback(null, student_instance);
-      }
-    });
-  },
-
-  createTutor: function(tutor, callback) {
-    Tutor.create(tutor, function (err, tutor_instance) {
-      if (err) {
-        console.error('Error creating a new tutor:', err);
-        callback(err);
-      } else {
-        callback(null, tutor_instance);
-      }
-    });
-  },
-
-  getUser: function(username, callback) {
+    },
+
+    createStudent: function(student, callback) {
+        Student.create(student, function (err, student_instance) {
+            if (err) {
+                console.error('Error creating a new student:', err);
+                callback(err);
+            } else {
+                callback(null, student_instance);
+            }
+        });
+    },
+
+    createTutor: function(tutor, callback) {
+
+        // TO DO : VALIDATE TUTOR BEFORE YOU SAVE IT
```

Comment:
Not sure if we should put this here or in a layer above. I feel like logical decisions should not be located here

Reply:
Ok, i see you're doing validation in a separate file, so this is fine.